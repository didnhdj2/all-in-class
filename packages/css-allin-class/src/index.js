import { getPlugin } from './utils/plugins.js'

export default function createPlugin(userConfig = {}) {
	return getPlugin(userConfig)
}

export function allin(userConfig = {}) {
	return getPlugin(userConfig)
}