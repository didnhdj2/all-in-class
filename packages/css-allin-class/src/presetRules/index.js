import { dynamicRules, staticRules } from "./styleSheet"

// let presetStaticRules = Object.values(staticRules).map(item => item.ailwindcss)

// TODO 
// 统计样式的大小 建议用户尽量公用
// 写自己的预设插件
// 主题
// 权重过低无法覆盖底部多层级的样式。
//   1.  id选择器
//   2.  import

// console.log('==== presetRules :', presetStaticRules);

function getrules(unit='px') {
	 return [
		
		// TODO 
		// 方法返回的值 可以是对象 可以是字符串
		// 去掉用户返回字符串的空格,有些空格不能去
		// ['jc-c',  'margin-left-right: auto;'],
		// ['jc-c',  'margin-left-right: auto;'],
		// ['mlr-auto',  'margin-left-right: auto;'],
		// ['m-0-auto',  'margin: 0 auto;'],
		// [/^m-(\d+)-(\d+)$/, match => ({ 'margin': `${match[1]}${unit} ${match[2]}${unit};` })],
		// [/^m-(\d+)$/, ([, d]) => ({ 'margin': `${d}${unit};` })],
		// [/^mt-(\d+)$/, ([, d]) => ({ 'margin-top': `${d}${unit};` })],
		// [/^mb-(\d+)$/, ([, d]) => ({ 'margin-bottom': `${d}${unit};` })],
		// [/^ml-(\d+)$/, ([, d]) => ({ 'margin-left': `${d}${unit};` })],
		// [/^mr-(\d+)$/, ([, d]) => ({ 'margin-right': `${d}${unit};` })],
		// [/^my-(\d+)$/, match =>  `margin-left:${match[1] }${unit};margin-right:${match[1]}${unit};`],
		// [/^px-(\d+)$/, match =>  `padding-top:${match[1] }${unit};padding-bottom:${match[1]}${unit};`],
		// [/^mlr-(\d+)$/, ([, d]) =>  `margin-top:${match[1] }${unit};margin-bottom:${match[1]}${unit};`],
		// [/^mtb-(\d+)$/, ([, d]) =>  `padding-top:${match[1] }${unit};padding-bottom:${match[1]}${unit};`],
		// [/^m-(\d+)-(\d+)-(\d+)-(\d+)$/, match => ({ 'margin': `${match[1]}${unit} ${match[2]}${unit} ${match[3]}${unit} ${match[4]}${unit};` })],
		
		// [/^p-(\d+)-(\d+)$/, match => ({ 'padding': `${match[1]}${unit} ${match[2]}${unit};` })],
		// [/^p-(\d+)$/, match => ({ 'padding': `${match[1]}${unit}` })],
		// [/^pt-(\d+)$/, match => ({ 'padding-top': `${match[1]}${unit}` })],
		// [/^pb-(\d+)$/, match => ({ 'padding-bottom': `${match[1]}${unit}` })],
		// [/^pl-(\d+)$/, match => ({ 'padding-left': `${match[1]}${unit}` })],
		// [/^pr-(\d+)$/, match => ({ 'padding-right': `${match[1]}${unit}` })],
		// [/^py-(\d+)$/, match =>  `padding-top:${match[1] }${unit};padding-bottom:${match[1]}${unit};`],
		// [/^px-(\d+)$/, match => `padding-left:${match[1] }${unit};padding-right:${match[1]}${unit};`],
		// [/^p-(\d+)-(\d+)-(\d+)-(\d+)$/, match => ({ 'padding': `${match[1]}${unit} ${match[2]}${unit} ${match[3]}${unit} ${match[4]}${unit};` })],
		
		// [/^bg-(\d+|[A-Za-z]+)$/, match => ({ 'background-color': `#${match[1]}` })],
		// [/^c-(\d{3}|\d{6})$/, match => ({ 'color': `#${match[1]}` })],
		// [/^c-([\dA-Za-z]{3}|[\dA-Za-z]{6})$/, match => ({ 'color': `#${match[1]}` })],
		
		
		// [/^(br|radius)-(\d+)$/, match => ({ 'border-radius': `${match[2]}${unit}` })],
		// [/^(br|radius)-(\d+)-(\d+)-(\d+)-(\d+)$/, match => ({ 'border-radius': `${match[2]}${unit} ${match[3]}${unit} ${match[4]}${unit} ${match[5]}${unit};` })],
		
		// [/^fs-(\d+)$/, ([, d]) => ({ 'font-size': `${d}${unit};` })],
		// 重复的key,最后一个会覆盖前面的值
		// {
		// 	'text-white': 'color:black ;',
		// 	'text-white': 'color:white ;',
		// },
	]
}
export let composition = [
	// 对象形式
	// {
	// 	// shortcuts to multiple utilities
	// 	'btn': 'py-2 px-4 font-semibold rounded-lg shadow-md',
	// 	'btn-green': 'text-white bg-green hover:bg-green-700',
	// 	// single utility alias
	// 	'red': 'text-red-100'
	// },
	// // 数组形式
	// ['btn', 'py-2 px-4 font-semibold rounded-lg',{PREFIX: 'cus'}],
	// ['card', 'm-20 p-20 bg-green',{PREFIX: 'cus'}],
	// // TODO 给当前对象内的样式单独增加前缀
	// {
	// 	PREFIX: 'cus',
	// 	btn: 'py-2 px-4 font-semibold rounded-lg',
	// 	card: 'p-10 px-4 font-semibold rounded-lg shadow-md',
	// },
	// // dynamic shortcuts
	// [/^btn-(.*)$/, ([, c]) => `bg-${c}-400 text-${c}-100 py-2 px-4 rounded-lg`],
	[/(.*)-i(\d*)$/, ([r, c, d],getValue) => {
		let val = getValue(c)
		return ()=>{
			return val
		}
	}],
]

export function presetRules(config={}) {
	// 规则的书写,是一个 头疼的事情
	// 把新建一套规则的成本降低
	
	// 列出有哪些常用css样式 styleSheet
	
	let allStyle = {}
	// for (let [name, { style }] of Object.entries(styleSheet)) {
	// 	allStyle = { ...allStyle, ...style }
	// }
	
	// 自动生成预设的规则文档
	// 预设规则的插件,提示代码书写规则
	
	let rules = getrules(config.unit)
	
	// TODO 静态规则单独放
	return {
		rules,
		composition
	}
}
