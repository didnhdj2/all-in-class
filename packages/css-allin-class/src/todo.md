# 待实现
- 把class名重新定义，减小css体积
- 把相同属性内容的css合并为同一个，减小css体积
- 提取公用内容，减小css体积
- 编写代码提示插件
- 兼容unocss的预设
- 配置文件改变自动更新
- 实现伪类
- 实现深度选择
- 使用单独的配置文件
- 在页面内直接写style样式，然后插件转换为class样式
- 提供自定义css规则的网页工具，生成自己css规则

# 已实现
- 不需要手动导入css，css文件默认不需要引入 
- 触发css的修改，内容生成在app.vue文件中
- 把每次修改文件输出css，改为虚拟的内容
- 添加一个css生成成功的提示
- 可以生成css作为预览