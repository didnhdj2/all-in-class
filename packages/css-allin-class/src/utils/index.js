import { PLUGIN_PREFIX } from "../constant"



const factory = (name) => (value) => Object.prototype.toString.call(value) === `[object ${name}]`
export const isNull = factory('Null')
export const isObject = factory('Object')
export const isArray = factory('Array')
export const isMap = factory('Map')
export const isSet = factory('Set')
export const isPromise = factory('Promise')
export const isAsyncFunction = factory('AsyncFunction')

export const isString = (value) => typeof value === 'string'
export const notNull = (value) => value !== null
export const isEmpty = (value) => value !== null || value !== undefined || value !== ''
export const isFunction = (value) => factory('Function')(value) || isAsyncFunction(value)

export function toArray(value = []) {
	return isArray(value) ? value : [value]
}

export const toNoRepeatArray = (value) => Array.from(new Set(value))

export function noop() { }

const consoleFactory = (name) => (msg) => console.warn(`[${PLUGIN_PREFIX}]`, msg)
export const warn = consoleFactory('warn')
export const log = consoleFactory('log')
export const error = consoleFactory('error')