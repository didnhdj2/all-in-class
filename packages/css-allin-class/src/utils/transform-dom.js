import { parse } from "@vue/compiler-dom";
import MagicString from 'magic-string'

// 修改class的内容
export function transformVueClass(source) {
	if (!source) return
	
	let code = source
	let classTokens = []
	
	let ms = new MagicString(code)

	// 获取到class的位置
	const ast = getAst(code)
	walkTree(ast, {
		script: (script) => {
			// 交给bable
			// console.log('==== script :', script);
		},
		style: (style) => {
			// 交给postcss

		},
		nodeTransform: (node) => {

		},
		propClass: (prop) => {
			const content = prop?.value?.content
			if (!content) return

			// 生成tokens
			const tokens = content.split(/\s+/g)
			classTokens.push(...tokens)
			// {
			// 	content,
			// 	tokens
			// }
			
			// .map((item, index1) => {
			// 	if (item == '%') {
			// 		let index = index1 + prop.loc.start.offset + 7
			// 		ms.overwrite(index, index + 1, '_')
			// 	}
			// });

			// 替换%
			content.split('').map((item, index1) => {
				if (item == '%') {
					let index = index1 + prop.loc.start.offset + 7
					ms.update(index, index + 1, '_')
				}
			});
			
		},
		propBindClass: (prop) => {
			let content = prop?.exp?.content
			if (!content) return

			// console.log('==== prop.exp.content :', prop.exp.content);
			// {header:iiii == 1}
			// header?'header':'headerheader'
		},
		comment: (node) => {

		}
	})
	
	
	// console.log('==== 1 :', ms.toString());
	// transformStyle
	// 判断class是否需要修改
	// 修改% 50% => 50_
	// 修改. 0.5 => 0_5

	// width w 都是宽度 合并
	
	if (ms.hasChanged()) {
		
	}
	
	// 修改class
	return {
		classTokens,
		code: ms.hasChanged()?ms.toString():code
	}
}

export function getAst(source) {
	let tree
	try {
		tree = parse(source)
	} catch (err) {

	}
	return tree
}

// 遍历树
function walkTree(node, options = {}) {
	if (!node) return

	if (node.type === 1 && node.tag === 'style') {
		options?.style(node)
	}
	if (node.type === 1 && node.tag === 'script') {
		options?.script(node)
	}

	options?.nodeTransform(node)

	if (node.props) {
		for (let prop of node.props) {

			// 动态class
			if (prop.type === 7 && prop.name === 'bind' && prop.arg && prop.arg.content === 'class') {

				options?.propBindClass(prop)
			}

			// 静态class
			if (prop.type === 6 && prop.name === 'class') {
				options?.propClass(prop)
			}

		}
	}

	// 注释
	if (node.type === 3) {
		options?.comment(node)
	}
	if (node.children) {
		for (let child of node.children) {
			walkTree(child, options)
		}
	}
	
}

// export async function transformers(
//   original,
//   id
// ) {
//   // if (original.includes(IGNORE_COMMENT))
//   //   return

//   // const transformers = (ctx.uno.config.transformers || []).filter(i => (i.enforce || 'default') === enforce)
//   // if (!transformers.length)
//   //   return

//   let code = original
//   let ms = new MagicString(code)
//   const maps = []
// 	// ms
//   // for (const t of transformers) {
//   //   if (t.idFilter) {
//   //     if (!t.idFilter(id))
//   //       continue
//   //   }
//   //   else if (!ctx.filter(code, id)) {
//   //     continue
//   //   }
//   //   await t.transform(ms, id, ctx)
//   //   if (ms.hasChanged()) {
//   //     code = ms.toString()
//   //     maps.push(ms.generateMap({ hires: true, source: id }) as EncodedSourceMap)
//   //     ms = new MagicString(code)
//   //   }
//   // }

//   if (code !== original) {
//     // ctx.affectedModules.add(id)
//     return {
//       code,
//       map: remapping(maps, () => null) as SourceMap,
//     }
//   }
// }
