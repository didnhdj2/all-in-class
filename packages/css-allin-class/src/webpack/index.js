class MyClass {
	#privateCount = 0; // 使用私有字段来实现私有变量

	constructor() {
		this.publicCount = 0;
		this.data = {
			modelsCssCache: {}, // 记录每个模块的css变动
			outputCssCache: {}, // 生成过的css缓存，用于输出css
			styleSheet: {
				staticRules: {},
				dynamicRules: {},
				dynamicCompRules: {}
			}, // 
			config: {},
			Plugin: {}
		}
		this.increment()
	}

	increment = () => {
		this.#privateCount++;
		this.publicCount++;
	}
	setPlugin(Plugin) {
		this.data.modelsCssCache = Plugin.modelsCssCache
		this.data.outputCssCache = Plugin.outputCssCache
		this.data.styleSheet = Plugin.styleSheet
		this.data.config = Plugin.config
		this.data.Plugin = Plugin
	}
	getPlugin() {
		return this.data
	}
}
export const data1 = (plugin) => {
	let tempPlugin = new MyClass()
	return function() {
		return tempPlugin
	}
}
export const data = new MyClass();
// function outerFunction() {
// 	let count = 0;

// 	function innerFunction() {
// 		count++;
// 		console.log(count);
// 	}

// 	return innerFunction;
// }

// var closure = outerFunction();
// closure(); // 输出1
// closure(); // 输出2
// closure(); // 输出3
export const Singleton1 = (() => {
  let instance;
  const createInstance = () => {
    // 创建实例的逻辑
    return {
			num: 0
		};
  };
  return {
    getInstance() {
      if (!instance) {
        instance = createInstance();
      }
      return instance;
    }
  };
})();

export const Singleton = Singleton1.getInstance()