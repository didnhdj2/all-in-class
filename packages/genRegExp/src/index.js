const ret    = require('ret');
const DRange = require('drange');
const types  = ret.types;


class Reg2str {
  /**
   * @constructor
   * @param {RegExp|string} regexp
   * @param {string} m
   */
  constructor(regexp, mode) {
    this._setDefaults(regexp);
    if (regexp instanceof RegExp) {
      this.ignoreCase = regexp.ignoreCase;
      this.multiline = regexp.multiline;
      regexp = regexp.source;

    } else if (typeof regexp === 'string') {
      this.ignoreCase = mode && mode.indexOf('i') !== -1;
      this.multiline = mode && mode.indexOf('m') !== -1;
    } else {
      throw Error('Expected a regexp or string');
    }

    this.tokens = ret(regexp);
  }


  /**
   * Checks if some custom properties have been set for this regexp.
   *
   * @param {Reg2str} Reg2str
   * @param {RegExp} regexp
   */
  _setDefaults(regexp) {
    // When a repetitional token has its max set to Infinite,
    // Reg2str won't actually generate a random amount between min and Infinite
    // instead it will see Infinite as min + 100.
    this.max = regexp.max != null ? regexp.max :
      Reg2str.prototype.max != null ? Reg2str.prototype.max : 100;

    // This allows expanding to include additional characters
    // for instance: Reg2str.defaultRange.add(0, 65535);
    this.defaultRange = regexp.defaultRange ?
      regexp.defaultRange : this.defaultRange.clone();

    if (regexp.randInt) {
      this.randInt = regexp.randInt;
    }
  }


  /**
   * Generates the random string.
   *
   * @return {string}
   */
  gen() {
    return this._gen(this.tokens, []);
  }


  /**
   * Generate random string modeled after given tokens.
   *
   * @param {Object} token
   * @param {Array.<string>} groups
   * @return {string}
   */
  _gen(token, groups) {
    let stack, str, n, i, l, code, expandedSet;
    switch (token.type) {
      case types.ROOT:
      case types.GROUP:
        // Ignore lookaheads for now.
        if (token.followedBy || token.notFollowedBy) { return ''; }

        // Insert placeholder until group string is generated.
        if (token.remember && token.groupNumber === undefined) {
          token.groupNumber = groups.push(null) - 1;
        }

        stack = token.options ?
          this._randSelect(token.options) : token.stack;
					
				
        str = '';
        for (i = 0, l = stack.length; i < l; i++) {
          str += this._gen(stack[i], groups);
        }

        if (token.remember) {
          groups[token.groupNumber] = str;
        }
				
        return str;

      case types.POSITION:
        // Do nothing for now.
        return '';

      case types.SET:
        expandedSet = this._expand(token);
        if (!expandedSet.length) { return ''; }
				let aa = this._randSelect(expandedSet)
				let resstr = String.fromCharCode(aa)
				// console.log('==== resstr :', resstr,aa,expandedSet);
        return resstr;
				
      case types.REPETITION:
        // Randomly generate number between min and max.
        n = this.randInt(token.min,
          token.max === Infinity ? token.min + this.max : token.max);

        str = '';
        for (i = 0; i < n; i++) {
          str += this._gen(token.value, groups);
        }

        return str;

      case types.REFERENCE:
        return groups[token.value - 1] || '';

      case types.CHAR:
        code = this.ignoreCase && this._randBool() ?
          this._toOtherCase(token.value) : token.value;
        return String.fromCharCode(code);
    }
  }


  /**
   * If code is alphabetic, converts to other case.
   * If not alphabetic, returns back code.
   *
   * @param {number} code
   * @return {number}
   */
  _toOtherCase(code) {
    return code + (97 <= code && code <= 122 ? -32 :
      65 <= code && code <= 90  ?  32 : 0);
  }


  /**
   * Randomly returns a true or false value.
   *
   * @return {boolean}
   */
  _randBool() {
    return !this.randInt(0, 1);
  }


  /**
   * Randomly selects and returns a value from the array.
   *
   * @param {Array.<Object>} arr
   * @return {Object}
   */
  _randSelect(arr) {
		
    if (arr instanceof DRange) {
      return arr.index(this.randInt(0, arr.length - 1));
    }
    return arr[this.randInt(0, arr.length - 1)];
  }


  /**
   * Expands a token to a DiscontinuousRange of characters which has a
   * length and an index function (for random selecting).
   *
   * @param {Object} token
   * @return {DiscontinuousRange}
   */
  _expand(token) {
    if (token.type === ret.types.CHAR) {
      return new DRange(token.value);
    } else if (token.type === ret.types.RANGE) {
			
      return new DRange(token.from, token.to);
    } else {
      let drange = new DRange();
      for (let i = 0; i < token.set.length; i++) {
        let subrange = this._expand(token.set[i]);
        drange.add(subrange);
        if (this.ignoreCase) {
          for (let j = 0; j < subrange.length; j++) {
            let code = subrange.index(j);
            let otherCaseCode = this._toOtherCase(code);
            if (code !== otherCaseCode) {
              drange.add(otherCaseCode);
            }
          }
        }
      }
			
      if (token.not) {
        return this.defaultRange.clone().subtract(drange);
      } else {
        return this.defaultRange.clone().intersect(drange);
      }
    }
  }


  /**
   * Randomly generates and returns a number between a and b (inclusive).
   *
   * @param {number} a
   * @param {number} b
   * @return {number}
   */
  randInt(a, b) {
    return a + Math.floor(Math.random() * (1 + b - a));
  }
	
	setMax(value) {
		this.max = Number(value)
	  return this;
	}

  /**
   * Default range of characters to generate from.
   */
  get defaultRange() {
    return this._range = this._range || new DRange(32, 126);
  }

  set defaultRange(range) {
    this._range = range;
  }


  /**
   *
   * Enables use of Reg2str with a shorter call.
   *
   * @param {RegExp|string| regexp}
   * @param {string} m
   * @return {string}
   */
  static Reg2str(regexp, m) {
    let Reg2str;
    if(typeof regexp === 'string') {
      regexp = new RegExp(regexp, m);
    }

    if (regexp._Reg2str === undefined) {
      Reg2str = new Reg2str(regexp, m);
      regexp._Reg2str = Reg2str;
    } else {
      Reg2str = regexp._Reg2str;
      Reg2str._setDefaults(regexp);
    }
    return Reg2str.gen();
  }


  /**
   * Enables sugary /regexp/.gen syntax.
   */
  static sugar() {
    /* eshint freeze:false */
    RegExp.prototype.gen = function() {
      return Reg2str.Reg2str(this);
    };
  }
};

exports.Reg2str = Reg2str