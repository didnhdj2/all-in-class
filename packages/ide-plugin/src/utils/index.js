import { PLUGIN_PREFIX } from "../constants"



const factory = (name) => {
	return (value) => Object.prototype.toString.call(value) === `[object ${name}]`
}
export const isNull = factory('Null')
export const isObject = factory('Object')
export const isArray = factory('Array')
export const isMap = factory('Map')
export const isSet = factory('Set')
export const isPromise = factory('Promise')
export const isAsyncFunction = factory('AsyncFunction')

export const isString = (value) => typeof value === 'string'
export const notNull = (value) => value !== null
export const isEmpty = (value) => value !== null || value !== undefined || value !== ''
export const isFunction = (value) =>  factory('Function')(value)||isAsyncFunction(value)

// :07:24.253 ==== fun : [AsyncFunction (anonymous)]
// 15:07:24.254 ==== isFunction(fun) : false
// 15:07:24.264 ==== Object.prototype.toString.call(fun) : [object AsyncFunction]
// 15:07:24.276 ==== pre : [AsyncFunction (anonymous)],

export function toArray(value = []) {
	return isArray(value) ? value : [value]
}

export const toNoRepeatArray = (value) => Array.from(new Set(value))

export function noop() {}


// const warned = new Set()
// export function warnOnce(msg) {
//   if (warned.has(msg)) return
//   console.warn('[odcss]', msg)
//   warned.add(msg)
// }

export function warn(msg) {
	console.warn(`[${PLUGIN_PREFIX}]`, msg)
}
export function error(msg) {
	console.error(`[${PLUGIN_PREFIX}]`, msg)
}
// export function mergeDeep<T>(original: T, patch: DeepPartial<T>): T {
//   const o = original as any
//   const p = patch as any

//   if (Array.isArray(o))
//     return [...p] as any

//   const output = { ...o }
//   if (isObject(o) && isObject(p)) {
//     Object.keys(p).forEach((key) => {
//       if (((isObject(o[key]) && isObject(p[key])) || (Array.isArray(o[key]) && Array.isArray(p[key]))))
//         output[key] = mergeDeep(o[key], p[key])
//       else
//         Object.assign(output, { [key]: p[key] })
//     })
//   }
//   return output
// }

// export function clone<T>(val: T): T {
//   let k: any, out: any, tmp: any

//   if (Array.isArray(val)) {
//     out = Array(k = val.length)
//     // eslint-disable-next-line no-cond-assign
//     while (k--) out[k] = ((tmp = val[k]) && typeof tmp === 'object') ? clone(tmp) : tmp
//     return out as any
//   }

//   if (Object.prototype.toString.call(val) === '[object Object]') {
//     out = {} // null
//     for (k in val) {
//       if (k === '__proto__') {
//         Object.defineProperty(out, k, {
//           value: clone((val as any)[k]),
//           configurable: true,
//           enumerable: true,
//           writable: true,
//         })
//       }
//       else {
//         // eslint-disable-next-line no-cond-assign
//         out[k] = ((tmp = (val as any)[k]) && typeof tmp === 'object') ? clone(tmp) : tmp
//       }
//     }
//     return out
//   }

//   return val
// }
