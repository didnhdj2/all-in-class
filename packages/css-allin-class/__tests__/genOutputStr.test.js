import { describe, assert, expect, test, it } from 'vitest'
import preset from '../src/presetRules/preset';
import { genCssValue, genStyle } from '../src/styles';
import { createStyleSheet } from '../src/styles/createStyleSheet';
import { genOutputStr } from '../src/utils/style';
import { transformVueClass } from '../src/utils/transform';
import { vue2, vue3 } from './const/vue';

const userConfig = {
	presets: [preset]
}
const styleSheet = await createStyleSheet({ userConfig })

describe('vue文件 生成css和修改vue文件', async () => {
	it('vue3', async ({ expect }) => {
		const outputCssCache = {}
		const id = 'file1'
		const { classTokens, code } = transformVueClass(vue3, id)
		const plugin = { styleSheet, outputCssCache, modelsCssCache: {}, userConfig }
		let changed = genStyle(classTokens, plugin, id) // 生成css
		expect(code).toMatchSnapshot()
		expect(genOutputStr({ outputCss: outputCssCache })).toMatchSnapshot()
	})

	it('vue2', async ({ expect }) => {
		const outputCssCache = {}
		const id = 'file1'
		const { classTokens, code } = transformVueClass(vue2, id)
		const plugin = { styleSheet, outputCssCache, modelsCssCache: {}, userConfig }
		let changed = genStyle(classTokens, plugin, id) // 生成css
		expect(code).toMatchSnapshot()
		expect(genOutputStr({ outputCss: outputCssCache })).toMatchSnapshot()
	})
})
