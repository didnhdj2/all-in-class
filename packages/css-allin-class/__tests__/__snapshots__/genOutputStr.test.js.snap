// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vue文件 生成css和修改vue文件 > vue2 1`] = `
"
<template>
  <div id=\\"app\\">
    <div class=\\"\\">
        <div class=\\"bg-_ff5500 c-_fff w-50_5_ ccc\\">11</div>
        <div class=\\"\\"  :class=\\"{'bg-_ff5500 c-_fff p-20px mt-10px w-100_05px': data.isPadding}\\">22</div>
        <div class=\\"\\"  :class=\\"data.isPadding ? 'bg-_ff5500 c-_fff p-20px mt-10px w-90_': ''\\">33</div>
    </div>
    <img alt=\\"Vue logo\\" src=\\"./assets/logo.png\\">
    <HelloWorld msg=\\"Welcome to Your Vue.js App\\"  :class=\\"\`fasdfasd: ''\`\\" />
		<WidthHeight></WidthHeight>
		<PaddingMargin></PaddingMargin>
  </div>
</template>

<script>
import HelloWorld from './components/HelloWorld.vue'
import WidthHeight from './components/WidthHeight.vue'
import PaddingMargin from './components/PaddingMargin.vue'

export default {
  name: 'App',
  components: {
    HelloWorld,
		WidthHeight,
		PaddingMargin
  }
}
</script>

<style>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>

"
`;

exports[`vue文件 生成css和修改vue文件 > vue2 2`] = `".bg-_ff5500{background-color:#ff5500;}.c-_fff{color:#fff;}.w-50_5_{width:50.5%;}.p-20px{padding:20px;}.mt-10px{margin-top:10px;}.w-100_05px{width:100.05px;}.w-90_{width:90%;}"`;

exports[`vue文件 生成css和修改vue文件 > vue3 1`] = `
"
<template>
	<div class=\\"\\">
		<div class=\\"bg-_ff5500 c-_fff w-50_5_ ccc\\">11</div>
		<div class=\\"\\"  :class=\\"{'bg-_ff5500 c-_fff p-20px mt-10px w-100_05px': data.isPadding}\\">22</div>
		<div class=\\"\\"  :class=\\"data.isPadding ? 'bg-_ff5500 c-_fff p-20px mt-10px w-90_': ''\\">33</div>
	</div>
</template>
<script setup>
	import { ref, reactive } from 'vue';

	const data = reactive({
		isPadding: true,
	})
</script>

<style>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>
"
`;

exports[`vue文件 生成css和修改vue文件 > vue3 2`] = `".bg-_ff5500{background-color:#ff5500;}.c-_fff{color:#fff;}.w-50_5_{width:50.5%;}.p-20px{padding:20px;}.mt-10px{margin-top:10px;}.w-100_05px{width:100.05px;}.w-90_{width:90%;}"`;
